beacon-chain/p2p/broadcaster_test.go

```diff
@@ -701,56 +702,44 @@ func TestService_BroadcastDataColumn(t *testing.T) {
 	}
 
 	// Create a listener.
-	listener, err := p.startDiscoveryV5(ipAddr, pkey)
+	listener, err := service.startDiscoveryV5(ipAddr, pkey)
 	require.NoError(t, err)
 
-	p.dv5Listener = listener
+	service.dv5Listener = listener
 
-	digest, err := p.currentForkDigest()
+	digest, err := service.currentForkDigest()
 	require.NoError(t, err)
 
 	subnet := peerdas.ComputeSubnetForDataColumnSidecar(columnIndex)
-	topic := fmt.Sprintf(topicFormat, digest, subnet)
+	topic := fmt.Sprintf(topicFormat, digest, subnet) + service.Encoding().ProtocolSuffix()
 
 	roSidecars, _ := util.CreateTestVerifiedRoDataColumnSidecars(t, []util.DataColumnParam{{Index: columnIndex}})
 	sidecar := roSidecars[0].DataColumnSidecar
 
-	// Async listen for the pubsub, must be before the broadcast.
-	var wg sync.WaitGroup
-	wg.Add(1)
-
-	peersChecked := make(chan bool, 0)
-
-	go func(tt *testing.T) {
-		defer wg.Done()
-
-		ctx, cancel := context.WithTimeout(t.Context(), 5*time.Second)
-		defer cancel()
-
-		// Wait for the peers to be checked.
-		<-peersChecked
+	// Attempt to broadcast nil object should fail.
+	var emptyRoot [fieldparams.RootLength]byte
+	err = service.BroadcastDataColumn(emptyRoot, subnet, nil)
+	require.ErrorContains(t, "attempted to broadcast nil", err)
 
-		// External peer subscribes to the topic.
-		topic += p.Encoding().ProtocolSuffix()
-		sub, err := p2.SubscribeToTopic(topic)
-		require.NoError(tt, err)
+	// Subscribe to the topic.
+	sub, err := p2.SubscribeToTopic(topic)
+	require.NoError(t, err)
 
-		msg, err := sub.Next(ctx)
-		require.NoError(tt, err)
+	// libp2p fails without this delay
+	time.Sleep(50 * time.Millisecond)
```

Comment:
Why do we have to do this? The prior test did not have a sleep and time.Sleep make tests flaky.

Reply:
We don't actually.
Fixed in 4c8a129090a9ef032ff0b63d1d5be4b9c80f35f9.

Reply:
Actually it seems we need.
There is this time.sleep for other equivalent tests in the same file, like [here](https://github.com/OffchainLabs/prysm/commit/b8e550b1e9d1146b2e6b71573de63968d9ba3328)

> The prior test did not have a sleep 
The real question is: How did this work without the `time.Sleep`?

Reply:
The prior test did not have a sleep but have a `<-peersChecked` which acted like a sleep.
That's why here a sleep is not needed.